pipeline/objects/model.py

```diff
@@ -22,3 +24,17 @@ def __init__(self, model: Any, *, name: str = "", local_id: str = None):
         self.source = inspect.getsource(model.__class__)
         self.hash = sha256(self.source.encode()).hexdigest()
         self.local_id = generate_id(10) if local_id is None else local_id
+        if not hasattr(self.model, "local_id"):
+            setattr(self.model, "local_id", self.local_id)
+
+    @classmethod
+    def from_schema(cls, schema: ModelGet):
+        pickled_data = hex_to_python_object(schema.hex_file.data)
+        if isinstance(pickled_data, Model):
+            pickled_data.local_id = schema.id
+            return pickled_data
+        return cls(
+            hex_to_python_object(schema.hex_file.data),
```

Comment:
Any need to repeat this call? Could you instead do

```suggestion
            pickled_data,
```